adam-core/src/main/scala/org/bdgenomics/adam/converters/FastqRecordConverter.scala

```diff
@@ -74,16 +77,52 @@ class FastqRecordConverter extends Serializable with Logging {
         .build())
   }
 
+  def convertFragment(element: (Void, Text)): Fragment = {
+    val lines = element._2.toString.split('\n')
+    require(lines.length == 8, "Record has wrong format:\n" + element._2.toString)
```

Comment:
I don't believe this is a fair assumption, and is most likely also a problem with FASTQ processing in ADAM generally.

Have a look at some pathological cases here
https://github.com/biojava/biojava/tree/master/biojava-sequencing/src/test/resources/org/biojava/nbio/sequencing/io/fastq